docs/architecture/notes.md

```diff
@@ -196,10 +195,32 @@ hash(hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash), vault_hash)
 !!! info
     To compute a note's hash, we do not need to know the note's `serial_num`. Knowing the hash of the `serial_num` (as well as `script_hash`, `input_hash` and `note_vault`) is also sufficient. We compute the hash of `serial_num` as `hash(serial_num, [0; 4])` to simplify processing within the VM._
 
-## Note discovery
+## Note discovery (note tags)
+
+Note discovery describes the process by which Miden clients find notes they want to consume. Miden clients can query the Miden node for notes carrying a certain note tag. Note tags are best-effort filters for notes registered on the network. They are lightweight values used to speed up queries. Clients can follow tags for specific use cases, such as swap scripts, or agree with a group of users on certain tags for their specific needs. Tags are also used by the operator to identify notes intended for network execution and include the corresponding information on how to execute them.
+
+```arduino
+0b009f4adc47857e2f6
+```
+
+The example note tag above indicates that the network operator (Miden node) executes the note against the account with the ID `0x09f4adc47857e2f6`. In this case, the note and the account against which it gets executed must be `public`.
+
+The two most signification bits of the note tag have the following interpretation:
+
+| Prefix | Execution hint | Target   | Allowed note type |
+| ------ | :------------: | :------: | :----------------:|
+| `0b00` | Network        | Specific | NoteType::Public  |
+| `0b01` | Network        | Use case | NoteType::Public  |
+| `0b10` | Local          | Any      | NoteType::Public  |
+| `0b11` | Local          | Any      | Any               |
+
+- Execution hint: Set to `Network` for network transactions. These notes will be validated and, if possible, consumed in a network transaction.
+- Target: Describes how to interpret the bits in the note tag. For tags with a specific target, the rest of the tag is interpreted as an `account_id`. For use case values, the meaning of the rest of the tag is not specified by the protocol and can be used by applications built on top of the rollup.
+- Note type: Describes the note's storage mode, either `public` or `private`.
+
+The following 30 bits can represent anything; from account IDs to use cases or any custom logic agreed upon.
```

Comment:
```suggestion
The following 30 bits can represent anything; from account IDs to use cases or any custom logic.
```

Reply:
the assumption is custom logic has already been agreed upon, you don't need to state it